zerver/openapi/openapi.py

```diff
@@ -126,11 +126,16 @@ def create_endpoints_dict(self) -> None:
         # such as email they must be substituted with proper regex.
 
         email_regex = r'([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})'
+        email_or_id_regex = r'(([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5}))|([0-9]*)'
         self._endpoints_dict = {}
         for endpoint in self._openapi['paths']:
             if '{' not in endpoint:
                 continue
             path_regex = '^' + endpoint + '$'
+            # Email or id arguments include email and
+            # end with id so find and replace them with
+            # email_or_id_regex
+            path_regex = re.sub(r'{[^}]*email[^}]*id}', email_or_id_regex, path_regex)
```

Comment:
What does this logic do?  I realize we have similar adjacent code but it's all pretty hacky and I'd prefer to not add more.

Reply:
This code was to match <code>email_or_id</code> param and replace it with <code> email_or_id</code> .I also feel that we can go without it as I don't find much use of it.

Reply:
I removed it .